### PR TITLE
#1801 Refactor standalone struct service. Get rid of unnecessary Indigo-worker creation.

### PR DESCRIPTION
**Pull request by Nitvex**

### CODE REVIEW

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts [hunk 1]
```diff
@@ -148,20 +150,45 @@ function mapWarningGroup(property: string) {
...
+  private readonly EE: EventEmitter = new EventEmitter()
 
   constructor(defaultOptions: StructServiceOptions) {
     this.defaultOptions = defaultOptions
+    this.worker = new IndigoWorker()
+    this.worker.onmessage = (e: MessageEvent<OutputMessage<string>>) => {
+      const message: OutputMessage<string> = e.data
+      if (message.type !== undefined) {
+        const event = messageTypeToEventMapping[message.type]
+        console.log(`emitted event ${event}`)
```
Comment (evgeniiFrolov1) [inline packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:182]: Could you remove `console.log` pls?

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts [hunk 2]
```diff
@@ -460,7 +490,10 @@ class IndigoService implements StructService {
         data: commandData
       }
 
-      worker.postMessage(inputMessage)
+      this.EE.removeListener('calculateCip', action)
```
Comment (evgeniiFrolov1) [inline packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:493]: What do you think about replacing string literal with `enum` 
for example :
```js
 this.EE.removeListener(ListenerType.CalculateCip, action)
```

**Review by evgeniiFrolov1 — CHANGES_REQUESTED** [review]

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts (hunk 2 above)
Comment (Nitvex) [reply packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:493]: Good proposal :) Will do

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts (hunk 1 above)
Comment (Nitvex) [reply packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:182]: Okay, I thought that was intentional.

**Review by evgeniiFrolov1 — APPROVED** [review]

**Pull request merged**